frontend/utils.ts

```diff
@@ -278,56 +315,113 @@ export const formatckBTCTransaccion = (
   canister: string,
   subNumber?: string,
 ): Transaction => {
-  const { timestamp, transfer } = ckBTCTransaction;
-  const trans = { status: OperationStatusEnum.Enum.COMPLETED } as Transaction;
-  transfer?.forEach((operation: any) => {
-    const value = operation.amount;
-    const amount = value.toString();
-    trans.to = (operation.to.owner as Principal).toString();
-    trans.from = (operation.from.owner as Principal).toString();
-
-    if (operation.to.subaccount.length > 0)
-      trans.toSub = `0x${subUint8ArrayToHex((operation.to.subaccount as [Uint8Array])[0])}`;
-    else trans.toSub = "0x0";
-
-    if (operation.from.subaccount.length > 0)
-      trans.fromSub = `0x${subUint8ArrayToHex((operation.from.subaccount as [Uint8Array])[0])}`;
-    else trans.fromSub = "0x0";
-
-    const subCheck = subNumber;
-    if (trans.from === principal && trans.fromSub === subCheck) {
-      trans.type = TransactionTypeEnum.Enum.SEND;
-    } else {
-      trans.type = TransactionTypeEnum.Enum.RECEIVE;
-    }
-
-    trans.canisterId = canister;
-    trans.symbol = symbol;
-    trans.amount = amount;
-    trans.idx = id.toString();
+  const { timestamp, transfer, mint, burn, kind } = ckBTCTransaction;
+  const trans = { status: OperationStatusEnum.Enum.COMPLETED, kind: kind } as Transaction;
+  if (kind === "mint")
+    mint.forEach(
```

Comment:
Lets add some comments here or separate the code a little to avoid so much code in 1 place.

Reply:
where are the comments?